manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/AbstractWorker.java

```diff
@@ -57,9 +63,25 @@ public T handleWork(Work work) {
         }
 
         // Fail when we've had enough
-        if (areRetriesExceeded(work, managedResource) || isTimeoutExceeded(work, managedResource)) {
+        boolean areRetriesExceeded = areRetriesExceeded(work, managedResource);
+        boolean isTimeoutExceeded = isTimeoutExceeded(work, managedResource);
+        if (areRetriesExceeded || isTimeoutExceeded) {
             managedResource.setStatus(FAILED);
             persist(managedResource);
+
+            ProvisioningFailureException failure;
+            if (areRetriesExceeded) {
+                failure = new ProvisioningMaxRetriesExceededException(String.format(RETRIES_FAILURE_MESSAGE,
```

Comment:
@manstis If the `status_message` was just a string on the resource, we could just `r.setStatusMessage` here.

Reply:
I don't favour that due to i18n ramifications.

IMO the use of Java exceptions is _intended_ to support programmatic try/catch/throw hiding `BridgeError`'s from developers. A lot of RHOSE's codebase has developers throw Java exceptions only to be caught and converted into `BridgeError`. The scenario here, for timeout/retries, bastardizes this concept - simply because an exception is needed for the recording mechanism I've implemented. This could lookup a `BridgeError` from the exception class and use that instead of instantiating an `Exception` only for it to be converted to a `BridgeError` in the `recordError(...)` method.